batchflow/models/tf/mobilenet.py

```diff
@@ -224,3 +227,229 @@ def block(cls, inputs, filters, residual=False, strides=1, expansion_factor=6, w
+            whether to include squeeze and excitation block
+        name : str
+            scope name
+
+        Returns
+        -------
+        tf.Tensor
+        """
+
+        if h_swish:
+            kwargs['activation'] = cls.h_swish # will not work in current se_block
+
+        num_filters = int(cls.num_channels(inputs, kwargs.get('data_format')) * expansion_factor * width_factor)
+        x = conv_block(inputs, 'cna Cna', [num_filters, num_filters], [1, 3], name='%s-exp' % name,
+                       strides=[1, strides], **kwargs)
+
+        if se_block:
+            x = cls.se_block(x, int(num_filters * 0.25), name='%s-se' % name, data_format=kwargs['data_format'],
+                             activation=[tf.nn.relu, cls.h_sigmoid])
+
+        x = conv_block(x, 'cn', filters, 1, name='%s-down' % name, **kwargs)
+
+        if residual:
+            x = inputs + x
+        return x
+
+
+_V3_SMALL_DEFAULT_BODY = [
+    dict(repeats=1, filters=16, expansion_factor=1, strides=2, kernel_size=3, h_swish=False, se_block=True),
+    dict(repeats=1, filters=24, expansion_factor=4.5, strides=2, kernel_size=3, h_swish=False, se_block=False),
+    dict(repeats=1, filters=24, expansion_factor=3.7, strides=1, kernel_size=3, h_swish=False, se_block=False,
+         residual=True),
+    dict(repeats=1, filters=40, expansion_factor=4, strides=2, kernel_size=5, h_swish=True, se_block=True),
+    dict(repeats=2, filters=40, expansion_factor=6, strides=1, kernel_size=5, h_swish=True, se_block=True,
+         residual=True),
+    dict(repeats=2, filters=48, expansion_factor=3, strides=1, kernel_size=5, h_swish=True, se_block=True),
+    dict(repeats=3, filters=96, expansion_factor=6, strides=2, kernel_size=5, h_swish=True, se_block=True),
+]
+
+class MobileNet_v3_small(MobileNet_v3):
+    """ MobileNet version 3 small architecture """
+    @classmethod
+    def default_config(cls):
+        config = MobileNet_v3.default_config()
+        config['initial_block'].update(dict(layout='cna', filters=16, kernel_size=3, strides=2, activation=cls.h_swish))
+        config['body'].update(dict(width_factor=1, layout=deepcopy(_V3_SMALL_DEFAULT_BODY)))
+        config['head'].update(dict(layout='cnavcacV', filters=[576, 1280, 2], pool_size=7, kernel_size=1,
+                                   activation=cls.h_swish, se_block=True))
+        return config
+
+    @classmethod
+    def head(cls, inputs, name='head', **kwargs):
+        """ The last network layers which produce predictions
+
+        Parameters
+        ----------
+        inputs : tf.Tensor
+            input tensor
+        name : str
+            scope name
+
+        Returns
+        -------
+        tf.Tensor
+        """
+        kwargs = cls.fill_params('head', **kwargs)
+        layout = kwargs.pop('layout')
+        filters = kwargs.pop('filters')
+        if kwargs.get('se_block'):
+            x = conv_block(inputs, layout=layout[:3], filters=filters[0], name=name, **kwargs)
```

Comment:
This makes head config very unclear and confusing. Consider splitting it.

Reply:
Both splitting layout in the head config to several and splitting head moving se_block to the body don't seem good to me. Please suggest a variant that seems less confusing to you.

Reply:
You could make `se_block` a dict with all the needed params: filters, layout, activations, etc

Reply:
done